docs/dev/tutorials/custom-aa-tutorial.md

```diff
@@ -334,35 +329,42 @@ function isValidSignature(bytes32 _hash, bytes memory _signature)
     }
 }
-
 ```
 
 ### Transaction validation
 
-Let's implement the validation process. It is responsible for validating the signature of the transaction and incrementing the nonce. Note, that there are some limitations on what this method is allowed to do. You can read more about them [here](../developer-guides/aa.md#limitations-of-the-verification-step).
+The transaction validation process is responsible for validating the signature of the transaction and incrementing the nonce. 
 
-To increment the nonce, you should use the `incrementNonceIfEquals` method of the `NONCE_HOLDER_SYSTEM_CONTRACT` system contract. It takes the nonce of the transaction and checks whether the nonce is the same as the provided one. If not, the transaction reverts. Otherwise, the nonce is increased.
+:::info
+- There are [some limitations](../developer-guides/aa.md#limitations-of-the-verification-step) on this function.
+:::
 
-Even though the requirements above allow the accounts to touch only their storage slots, accessing your nonce in the `NONCE_HOLDER_SYSTEM_CONTRACT` is a [whitelisted](../developer-guides/aa.md#extending-the-set-of-slots-that-belong-to-a-user) case, since it behaves in the same way as your storage, it just happened to be in another contract. To call the `NONCE_HOLDER_SYSTEM_CONTRACT`, you should add the following import:
+To increment the nonce, use the `incrementNonceIfEquals` function from the `NONCE_HOLDER_SYSTEM_CONTRACT` system contract. It takes the nonce of the transaction and checks whether it is the same as the provided one. If not, the transaction reverts; otherwise, the nonce increases.
 
-```solidity
-// Access zkSync system contracts, in this case for nonce validation vs NONCE_HOLDER_SYSTEM_CONTRACT
-import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
-```
+Even though the requirements above mean the accounts only touch their own storage slots, accessing your nonce in the `NONCE_HOLDER_SYSTEM_CONTRACT` is a [whitelisted](../developer-guides/aa.md#extending-the-set-of-slots-that-belong-to-a-user) case, since it behaves in the same way as your storage, it just happens to be in another contract. 
 
-Note that since the non-view methods of the `NONCE_HOLDER_SYSTEM_CONTRACT` are required to be called with the `isSystem` flag on, the [systemCallWithPropagatedRevert](https://github.com/matter-labs/v2-testnet-contracts/blob/main/l2/system-contracts/libraries/SystemContractsCaller.sol#L75) method of the `SystemContractsCaller` library should be used, so this library needs to be imported as well:
+To call the `NONCE_HOLDER_SYSTEM_CONTRACT`, we add the following import:
 
 ```solidity
-// to call non-view method of system contracts
-import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";
+// Access zkSync system contracts, in this case for nonce validation vs NONCE_HOLDER_SYSTEM_CONTRACT
+import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
```

Comment:
@kailash-manasarovar this statement/comment isn't clear.
Is there suppose to be a link here?

Reply:
No link, **vs** should be **via**, can you suggest it please @krofax ?

Reply:
If you look two paragraphs up on the first entry, they explain it. I've changed `vs` to `via`.